### PR TITLE
ci: replace telegram notify with vk teams notify

### PR DESCRIPTION
**Pull request by lastoCHka42**
Resolves tarantool/tarantool-qa#258

### CODE REVIEW

File: .github/workflows/centos_7.yml [hunk 1]
```diff
@@ -54,18 +54,33 @@ jobs:
...
           OS: 'el'
           DIST: '7'
           GC64: ${{ matrix.build-type == 'gc64' }}
         uses: ./.github/actions/pack_and_deploy
-      - name: call action to send Telegram message on failure
-        env:
-          TELEGRAM_TOKEN: ${{ secrets.TELEGRAM_CORE_TOKEN }}
-          TELEGRAM_TO: ${{ secrets.TELEGRAM_CORE_TO }}
-        uses: ./.github/actions/send-telegram-notify
+      - name: Set chat for reports
```
Comment (ylobankov) [inline .github/workflows/centos_7.yml:64]: It looks like your approach has some downsides and the main one is code duplication. What about creating an internal composite action, say `report-job-status`, and moving this logic to it? Look at this:

```
- name: Send VK Teams message on failure
  if: failure()
  uses: ./.github/actions/report-job-status
  with:
    job-steps: ${{ ToJson(steps) }}
  env:
    VKTEAMS_API_URL: ${{ secrets.VKTEAMS_API_URL }}
    VKTEAMS_BOT_TOKEN: ${{ secrets.VKTEAMS_BOT_TOKEN }}
    VKTEAMS_CHAT_ID: ${{ secrets.VKTEAMS_CHAT_ID }}
```

or

```
- name: Send VK Teams message on failure
  if: failure()
  uses: ./.github/actions/report-job-status
  with:
    api-url: ${{ secrets.VKTEAMS_API_URL }}
    bot-token: ${{ secrets.VKTEAMS_BOT_TOKEN }}
    chat-id: ${{ secrets.VKTEAMS_CHAT_ID }}
    job-steps: ${{ ToJson(steps) }}
```

**Review by ylobankov — CHANGES_REQUESTED** [review]

File: .github/workflows/centos_7.yml (hunk 1 above)
Comment (lastoCHka42) [reply .github/workflows/centos_7.yml]: Done

File: .github/actions/report-job-status/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+(`master`, `2.10`, `1.10`)
+* Personal chat, created by committer – for any PR or for any event in other
+branches
+
+## How to create personal chat
+
+1. Create chat in VK Team, make it public (access can be limited by enabling 
+"Join in Approval" in chat settings)
+2. Change URL of the chat: use URL of main VK Teams chat and add
+"_<your_github_nickname>", example: `main_chat_mynickname`
```
Comment (ylobankov) [inline .github/actions/report-job-status/README.md:15]: for example: `https://vkteams.example.com/main_chat_id_mynickname`

File: .github/actions/report-job-status/action.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+name: 'Send VK Teams message on failure'
```
Comment (ylobankov) [inline .github/actions/report-job-status/action.yml:1]: It should be the same as in the README.md file.

File: .github/actions/report-job-status/action.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    description: steps context in JSON
+  api-url:
+    required: true
+    description: VK Teams API URL
+  bot-token:
+    required: true
+    description: VK Teams bot token
+  chat-id:
+    required: true
+    description: VK Teams main chat id
```
Comment (ylobankov) [inline .github/actions/report-job-status/action.yml:16]: id -> ID

File: .github/actions/report-job-status/action.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
```
Comment (ylobankov) [inline .github/actions/report-job-status/action.yml:17]: Too many blank lines.

File: .github/actions/report-job-status/action.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+runs:
+  using: "composite"
```
Comment (ylobankov) [inline .github/actions/report-job-status/action.yml:20]: Use single quotes or don't use them at all.

File: .github/actions/report-job-status/action.yml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+              $GITHUB_REF == 'refs/heads/2.10' ||
+              $GITHUB_REF == 'refs/heads/1.10' ]]; then
+          echo CHAT_ID="${{ inputs.chat-id }}" | tee -a "$GITHUB_ENV"
+        else 
+          echo CHAT_ID="${{ inputs.chat-id }}_${{ github.actor }}" \
+          | tee -a "$GITHUB_ENV"
+        fi
+
+    - name: Send VK Teams message on failure
+      shell: bash
```
Comment (ylobankov) [inline .github/actions/report-job-status/action.yml:38]: Looks like it is redundant.

File: .github/actions/report-job-status/action.yml [hunk 8]
```diff
@@ -0,0 +1,45 @@
+name: 'Send VK Teams message on failure'
+description: 'Send message to VK Teams channel/bot on failure'
+
+inputs:
+  job-steps:
+    required: true
+    description: steps context in JSON
```
Comment (ylobankov) [inline .github/actions/report-job-status/action.yml:7]: Steps

**Review by ylobankov — CHANGES_REQUESTED** [review]

File: .github/actions/report-job-status/README.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+Action uses[`report-job-status action`](https://github.com/tarantool/actions/tree/master/report-job-status)
+and change the chat for notifications:
+* Main chat (Tarantool CI/CD reports) - if the job started for stable branch
+(`master`, `2.10`, `1.10`)
+* Personal chat, created by committer – for any PR or for any event in other
+branches
+
+## How to create personal chat
+
+1. Create chat in VK Team, make it public (access can be limited by enabling
```
[the remark this answers is not in the document]
Comment (lastoCHka42) [reply .github/actions/report-job-status/README.md]: Yes, to be able to change the link

File: .github/actions/report-job-status/action.yml (hunk 3 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/action.yml:1]: Done

File: .github/actions/report-job-status/action.yml (hunk 8 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/action.yml:7]: Done

File: .github/actions/report-job-status/action.yml (hunk 4 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/action.yml:16]: Done

File: .github/actions/report-job-status/action.yml (hunk 5 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/action.yml:17]: Fixed

File: .github/actions/report-job-status/action.yml (hunk 6 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/action.yml:20]: Done

File: .github/actions/report-job-status/README.md (hunk 2 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/README.md:15]: Done

File: .github/actions/report-job-status/README.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
+# Send VK Teams notifications on failures
+
+Action uses[`report-job-status action`](https://github.com/tarantool/actions/tree/master/report-job-status)
+and change the chat for notifications:
```
[the remark this answers is not in the document]
Comment (lastoCHka42) [reply .github/actions/report-job-status/README.md:4]: Done

File: .github/actions/report-job-status/README.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
+# Send VK Teams notifications on failures
+
+Action uses[`report-job-status action`](https://github.com/tarantool/actions/tree/master/report-job-status)
```
[the remark this answers is not in the document]
Comment (lastoCHka42) [reply .github/actions/report-job-status/README.md]: Done

File: .github/actions/report-job-status/README.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+and change the chat for notifications:
+* Main chat (Tarantool CI/CD reports) - if the job started for stable branch
```
[the remark this answers is not in the document]
Comment (lastoCHka42) [reply .github/actions/report-job-status/README.md]: Done

File: .github/actions/report-job-status/action.yml [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+    description: VK Teams main chat ID
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Set chat for reports
+      run: |
+        # Report failures on commits in stable branches to the common chat;
+        # report failures in PRs to individual chats
+        if ${{ github.ref == 'refs/heads/master' ||
```
Comment (ylobankov) [inline .github/actions/report-job-status/action.yml:25]: Try this condition to cover all branches:

```
if [[ "${{ github.ref }}" == "refs/heads/master" || 
      "${{ github.ref }}" =~ refs/heads/[0-9].[0-9]+$ ]]; then
  ...
else
  ...
fi
```

File: .github/actions/report-job-status/README.md [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+(`master`, `2.10`, `1.10`)
+* Personal chat, created by committer – for any PR or for any event in other
```
[the remark this answers is not in the document]
Comment (lastoCHka42) [reply .github/actions/report-job-status/README.md]: Done

File: .github/actions/report-job-status/README.md [hunk 15]
```diff
@@ -0,0 +1,32 @@
+# Send VK Teams notifications on failures
+
+This action uses[`report-job-status action`](https://github.com/tarantool/actions/tree/master/report-job-status)
```
Comment (ylobankov) [inline .github/actions/report-job-status/README.md:3]: Backticks are not needed.

File: .github/actions/report-job-status/README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (ylobankov) [reply .github/actions/report-job-status/README.md]: Ok, got it.

File: .github/actions/report-job-status/README.md [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+## How to use Github Action from Github workflow
+
+Add the following code to the running steps:
+```yml
+  - name: Send VK Teams message on failure
+    if: failure()
+    uses: ./.github/actions/report-job-status
+    with:
+      job-steps: ${{ ToJson(steps) }}
+    env:
```
Comment (ylobankov) [inline .github/actions/report-job-status/README.md:28]: ???

File: .github/actions/report-job-status/README.md [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+## How to create personal chat
+
+1. Create new chat in VK Team, make it public (access can be limited by enabling
+"Join in Approval" in chat settings)
+2. Change the URL of the chat: use URL of the main VK Teams chat and add
+"_<your_github_nickname>", example:
+`https://vkteams.example.com/profile/main_chat_mynickname`
+3. Add the bot from the main chat to your personal chat
+
+## How to use Github Action from Github workflow
```
Comment (ylobankov) [inline .github/actions/report-job-status/README.md:19]: 1. GitHub
2. action

File: .github/actions/report-job-status/README.md [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+This action uses[`report-job-status action`](https://github.com/tarantool/actions/tree/master/report-job-status)
+and chooses the proper chat for notifications:
+* The main chat (Tarantool CI/CD reports) - if the job started for the stable
+branches (`2.10`, `1.10`) and `master`
+* A personal chat, created by committer – if the job started on creating/updating
+a pull request or for any event in other branches
+
+## How to create personal chat
+
+1. Create new chat in VK Team, make it public (access can be limited by enabling
```
Comment (ylobankov) [inline .github/actions/report-job-status/README.md:12]: 1. a new chat
2. VK Teams

File: .github/actions/report-job-status/README.md (hunk 15 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/README.md:3]: Done

File: .github/actions/report-job-status/action.yml (hunk 13 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/action.yml]: Done

File: .github/actions/report-job-status/README.md [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+* Main chat (Tarantool CI/CD reports) - if the job started for stable branch
+(`master`, `2.10`, `1.10`)
+* Personal chat, created by committer – for any PR or for any event in other
+branches
+
+## How to create personal chat
+
+1. Create chat in VK Team, make it public (access can be limited by enabling 
+"Join in Approval" in chat settings)
+2. Change URL of the chat: use URL of main VK Teams chat and add
```
[the remark this answers is not in the document]
Comment (lastoCHka42) [reply .github/actions/report-job-status/README.md]: Done

File: .github/actions/report-job-status/README.md (hunk 16 above)
Comment (ylobankov) [reply .github/actions/report-job-status/README.md]: Looks like fixed.

File: .github/actions/report-job-status/README.md (hunk 17 above)
Comment (ylobankov) [reply .github/actions/report-job-status/README.md]: Looks like fixed.

File: .github/actions/report-job-status/README.md (hunk 18 above)
Comment (ylobankov) [reply .github/actions/report-job-status/README.md]: Looks like fixed.

File: .github/actions/report-job-status/README.md [hunk 20]
```diff
@@ -0,0 +1,31 @@
+# Send VK Teams notification on failure
+
+This action uses [report-job-status action](https://github.com/tarantool/actions/tree/master/report-job-status)
+and chooses the proper chat for the notification:
```
Comment (ylobankov) [inline .github/actions/report-job-status/README.md:4]: ... for the notification on failure:

File: .github/actions/report-job-status/README.md [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+* The team chat (Tarantool CI/CD reports) - if the job started for the stable
```
Comment (ylobankov) [inline .github/actions/report-job-status/README.md:5]: ... - if the job started on push to the `master` or release branch (`1.10`, `2.10`, etc)

File: .github/actions/report-job-status/README.md [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+branches (`2.10`, `1.10`) and `master`
+* A personal chat, created by the committer – if the job started on
+creating/updating a pull request or for any event in other branches
```
Comment (ylobankov) [inline .github/actions/report-job-status/README.md:8]: ~~for~~ any event in other branches

File: .github/actions/report-job-status/action.yml [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+  chat-id:
+    required: true
+    description: VK Teams team chat ID
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Set chat for reports
+      run: |
+        # Report failures occurred in the 'master' and stable branches to
```
Comment (ylobankov) [inline .github/actions/report-job-status/action.yml:23]: ... and release branches ...

File: .github/actions/report-job-status/action.yml [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+        # the team chat. Report failures occurred in PRs and in other branches
```
Comment (ylobankov) [inline .github/actions/report-job-status/action.yml:24]: ... ~~in~~ other branches ...

File: .github/actions/report-job-status/action.yml [hunk 25]
```diff
@@ -0,0 +1,41 @@
+name: 'Send VK Teams notification on failure'
+description: 'Send the message to VK Teams chat on failure'
+
+inputs:
+  job-steps:
+    required: true
+    description: Steps context in JSON
```
[the remark this answers is not in the document]
Comment (lastoCHka42) [reply .github/actions/report-job-status/action.yml:7]: It's really can be empty, fixed

File: .github/actions/report-job-status/README.md (hunk 20 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/README.md:4]: done

File: .github/actions/report-job-status/README.md (hunk 21 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/README.md:5]: For current workflows, the triggered event also can be `scheduled`, theoretically it also can be many other: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows. 
stable -> release fixed

File: .github/actions/report-job-status/README.md (hunk 22 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/README.md:8]: done

File: .github/actions/report-job-status/action.yml (hunk 23 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/action.yml:23]: done

File: .github/actions/report-job-status/action.yml (hunk 24 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/action.yml:24]: done

File: .github/actions/report-job-status/action.yml [hunk 26]
```diff
@@ -0,0 +1,41 @@
+name: 'Send VK Teams notification on failure'
+description: 'Send the message to VK Teams chat on failure'
+
+inputs:
+  job-steps:
```
Comment (ylobankov) [inline .github/actions/report-job-status/action.yml:5]: Could you please make this param the last one?

File: .github/actions/report-job-status/action.yml [hunk 27]
```diff
@@ -0,0 +1,41 @@
+name: 'Send VK Teams notification on failure'
```
Comment (ylobankov) [inline .github/actions/report-job-status/action.yml:1]: Here you use quotes, but you don't use them below in param descriptions. You should choose one option.

File: .github/actions/report-job-status/README.md [hunk 28]
```diff
@@ -0,0 +1,31 @@
+# Send VK Teams notification on failure
+
+This action uses [report-job-status action](https://github.com/tarantool/actions/tree/master/report-job-status)
+and chooses the proper chat for the notification on failure:
+* The team chat (Tarantool CI/CD reports) - if the job started on `master` and
```
Comment (ylobankov) [inline .github/actions/report-job-status/README.md:5]: ... - if the job started on any event in the `master` or release branch (`1.10`, `2.10`, etc)

File: .github/actions/report-job-status/action.yml (hunk 26 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/action.yml:5]: done

File: .github/actions/report-job-status/README.md (hunk 21 above)
Comment (ylobankov) [reply .github/actions/report-job-status/README.md]: Ok, got it.

File: .github/actions/report-job-status/action.yml (hunk 27 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/action.yml:1]: done

File: .github/actions/report-job-status/README.md (hunk 28 above)
Comment (lastoCHka42) [reply .github/actions/report-job-status/README.md:5]: done

File: .github/actions/report-job-status/README.md [hunk 29]
```diff
@@ -0,0 +1,31 @@
+# Send VK Teams notification on failure
+
+This action uses [report-job-status action](https://github.com/tarantool/actions/tree/master/report-job-status)
```
Comment (ylobankov) [inline .github/actions/report-job-status/README.md:3]: I suggest using this variant:

`This action selects the corresponding chat for the notification on failure and uses [report-job-status action](https://github.com/tarantool/actions/tree/master/report-job-status) to send it.`

File: .github/actions/report-job-status/README.md [hunk 30]
```diff
@@ -0,0 +1,31 @@
...
+and chooses the corresponding chat for the notification on failure:
+* The team chat (Tarantool CI/CD reports) - if the job started on any event in
+the `master` or release branches (`2.10`, `1.10`, etc)
+* A personal chat, created by the committer – if the job started on
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply .github/actions/report-job-status/README.md:8]: Looks like fixed.

File: .github/actions/report-job-status/README.md [hunk 31]
```diff
@@ -0,0 +1,31 @@
...
+creating/updating a pull request or any event in other branches
+
+## How to create personal chat
+
+1. Create a new public chat in VK Teams (access can be limited by enabling
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply .github/actions/report-job-status/README.md:13]: Looks like fixed.

File: .github/actions/report-job-status/README.md [hunk 32]
```diff
@@ -0,0 +1,31 @@
...
+"Join in Approval" in the chat settings)
+2. Change URL of the created chat: use URL of the team chat and add
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply .github/actions/report-job-status/README.md:15]: Looks like fixed.

File: .github/actions/report-job-status/README.md (hunk 29 above)
Comment (ylobankov) [reply .github/actions/report-job-status/README.md]: Looks like fixed.

File: .github/actions/report-job-status/README.md [hunk 33]
```diff
@@ -0,0 +1,31 @@
...
+`_<your_github_nickname>` to it, for example:
+`https://vkteams.example.com/profile/team_chat_mynickname`
+3. Add the bot from the team chat to your personal chat
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply .github/actions/report-job-status/README.md]: Looks like fixed.

File: .github/actions/report-job-status/README.md [hunk 34]
```diff
@@ -0,0 +1,31 @@
...
+and chooses the corresponding chat for the notification on failure:
+* The team chat (Tarantool CI/CD reports) - if the job started on any event in
+the `master` or release branches (`2.10`, `1.10`, etc)
+* A personal chat, created by the committer – if the job started on
+creating/updating a pull request or any event in other branches
+
+## How to create personal chat
+
+1. Create a new public chat in VK Teams (access can be limited by enabling
+"Join in Approval" in the chat settings)
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply .github/actions/report-job-status/README.md]: Looks like fixed.

File: .github/actions/report-job-status/README.md [hunk 35]
```diff
@@ -0,0 +1,31 @@
+# Send VK Teams notification on failure
+
+This action uses [report-job-status action](https://github.com/tarantool/actions/tree/master/report-job-status)
+and chooses the corresponding chat for the notification on failure:
+* The team chat (Tarantool CI/CD reports) - if the job started on any event in
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply .github/actions/report-job-status/README.md]: Looks like fixed.

File: .github/workflows/centos_7.yml [hunk 36]
```diff
@@ -60,12 +60,14 @@ jobs:
...
-          TELEGRAM_TO: ${{ secrets.TELEGRAM_CORE_TO }}
-        uses: ./.github/actions/send-telegram-notify
+      - name: Send VK Teams message on failure
         if: failure()
+        uses: ./.github/actions/report-job-status
+        with:
+          api-url: ${{ secrets.VKTEAMS_API_URL }}
+          bot-token: ${{ secrets.VKTEAMS_BOT_TOKEN }}
+          chat-id: ${{ secrets.VKTEAMS_CHAT_ID }}
+          job-steps: ${{ ToJson(steps) }}
```
Comment (ylobankov) [inline .github/workflows/centos_7.yml:70]: Looks like this is not needed.

File: .github/workflows/coverage.yml [hunk 37]
```diff
@@ -48,36 +48,30 @@ jobs:
         with:
           fetch-depth: 0
           submodules: recursive
-
```
Comment (ylobankov) [inline .github/workflows/coverage.yml:51]: Please revert these lines back.

File: .github/workflows/debug.yml [hunk 38]
```diff
@@ -48,23 +48,19 @@ jobs:
         with:
           fetch-depth: 0
           submodules: recursive
-
```
Comment (ylobankov) [inline .github/workflows/debug.yml:51]: Please revert these lines back.

File: .github/workflows/fuzzing.yml [hunk 39]
```diff
@@ -52,8 +52,8 @@ jobs:
 
     steps:
       - name: build fuzzers (${{ matrix.sanitizer }})
-        id: build
         uses: google/oss-fuzz/infra/cifuzz/actions/build_fuzzers@master
+        id: build
```
Comment (ylobankov) [inline .github/workflows/fuzzing.yml:56]: Why did you move this line?

Comment (lastoCHka42) [reply .github/workflows/fuzzing.yml]: fixed

File: .github/actions/report-job-status/README.md [hunk 40]
```diff
@@ -0,0 +1,31 @@
...
+creating/updating a pull request or any event in other branches.
+
+## How to create personal chat
+
+1. Create a new public chat in VK Teams (access can be limited by enabling
+`Join in Approval` in the chat settings).
+2. Change URL of the created chat: use URL of the team chat and add
+`_<your_github_nickname>` to it, for example:
+`https://vkteams.example.com/profile/team_chat_mynickname`.
+3. Add the bot from the team chat to your personal chat.
```
Comment (ylobankov) [inline .github/actions/report-job-status/README.md:18]: I think it is worth mentioning the user nickname:

Add the bot user `tarantoolbot` to your personal chat.

**Review by ylobankov — APPROVED** [review]
Now LGTM, thanks!

File: .github/actions/report-job-status/README.md (hunk 40 above)
Comment (ylobankov) [reply .github/actions/report-job-status/README.md]: Thanks, looks like fixed.
